### ISSUE TITLE
save many-to-none reference to db

### ISSUE THREAD

**Issue by dani3lsz** [rmmmp/ember-cloud-firestore-adapter#66]
Hi,

First of all, thanks for the great tool!!

I am trying to add a child to a parent doc with a many-to-none relationship between them. But both object exists before I create the relationship. (I use version 0.5.0 of the adapter)

Parent model:
```
children: hasMany('child', {
    inverse: null,
    buildReference(db, record) {
      return db.collection('parents').doc(record.get('id')).collection('children');
    }
  }),
```

The child model doesn't have a reference to the parent, because I don't need that.

When a user adds a child to a parent, I am doing: 
```
parent.get('children').pushObject(child);
```

My question is, how do I persist this in the database?
If I do `parent.save()` it doesn't create a subcollection reference to the child. Also `parent.get('hasDirtyAttributes')` returns `false`.

Maybe I just don't find this in the docs. In that case sorry for the inconvenience, but could you direct me to a solution?

Thanks in advance!

**Comment by dani3lsz** [comment]
Ok, I found a solution.

I made a model called reference, that looks like this:
```
cloudFirestoreReference: attr('string')
```

Then when the user adds a child to the parent I do this:
```
        this.get('store').createRecord('reference',{
          id: child.get('id'),
          cloudFirestoreReference: 'parents/' + child.get('id')
        }).save({
          adapterOptions: {
            buildReference(db) {
              return db.collection('parents').doc(parent.get('id')).collection('children');
            }
          }
        })
```

This way I can add/remove the reference quite easily.

Let me know if there is a simple method. Otherwise, just close this issue. Thanks again :)!

**Comment by dani3lsz** [comment]
So before you close this issue, I thought deleting will be easy, but it's not working it seems.

I try to do this to remove a reference:
First I get the reference object
```
    this.get('store').findRecord('reference', child.get('id'), {
        adapterOptions: {
          buildReference(db) {
            return db.collection('parents').doc(parent.get('id')).collection('children');
          }
        }
      })
```

Which I tried to delete with simply doing `.then(ref => {ref.destroyRecord()})` or 
```
.then(ref => {
  ref.deleteRecord();
  ref.save({
        adapterOptions: {
          buildReference(db) {
            return db.collection('parents').doc(parent.get('id')).collection('children');
          }
        }
      })
})
```

In both case I get this error:
```
Error: Attempted to handle event `pushedData` on <reference:my3QBGOqhqIUss8q6nvc> while in state root.deleted.inFlight. 
```

How should I delete a reference in a subcollection?

**Comment by rmmmp** [comment]
You'll have to manually persist the relationship yourself when it's a many-to-many or many-to-none. See the note [here](https://github.com/rmmmp/ember-cloud-firestore-adapter/blob/master/guides/04-creating-updating-deleting-records.md#batched-writes).

**Comment by rmmmp** [comment]
Also just to add. In your case, you don't need to implement a `buildReference` hook in the `hasMany` of your `parent` model. When the relationship is many-to-many or many-to-none, by default it will look for it under the sub-collection.

```javascript
children: hasMany('child', {
  inverse: null,
}),
```

Just having that setup above will default to `parents/<parent_id>/childrens` sub-collection.

**Comment by dani3lsz** [comment]
Thanks for the fast reply! I somehow looked over the batched writes.

Yeah, the deletion works fine that way. 

Also just a note, that could go into the guides maybe. As you define the full doc reference on batch.delete, I also had to do that to create the reference.

Like this: 
```
parent.save({
  adapterOptions: {
    include(batch, db) {
      batch.set(db.collection('parents').doc(parent.get('id')).collection('children').doc(child.get('id'),{});
    }
  }
});
```
Otherwise it gave an error. I guess it was because the parent was already existing in the db? Not sure. I also noticed that it doesn't matter what I try to save to create the relationship. To have the same result I can do
```
children.save({
  adapterOptions: {
    include(batch, db) {
      batch.set(db.collection('parents').doc(parent.get('id')).collection('children').doc(child.get('id'),{});
    }
  }
});
```
Which makes me wonder if I could just call the batch.set somehow directly, because to create the relationship, I actually don't need to save neither the parent or the child. 

And finally, after creating the reference in the way you see above, in the database, the subcollection will only have an id with no content. Well I don't know if this case I should see the `cloudFirestoreReference` automatically there or not. But even if it's not there, the adapter still finds the children just fine with the id only.

Thanks again!
